lib/SILOptimizer/IPO/DeadFunctionElimination.cpp

```diff
@@ -468,6 +468,33 @@ class ExternalFunctionDefinitionsElimination : FunctionLivenessComputation {
     /// Therefore there is no need for a liveness computation.
     /// The next line can be just replaced by:
     /// return false;
+
+    // Keep all transparent functions alive because we have to generate code for
+    // them (no code is generated for transparent functions in the module from
+    // which they are imported, e.g. from the stdlib).
+    // This only affects transparent functions which could not be inlined for
+    // some reason, e.g. functions which are referenced but not called.
+    // Here we handle the special case if a transparent function is referenced
+    // from a non-externally-available function (i.e. a function for which we
+    // generate code). And this function is only reachable through a
+    // vtable/witness-table. In such a case we would not visit the transparent
+    // function in findAliveFunctions() because we don't consider vtables/
+    // witness-tables as anchors.
+    for (SILFunction &F : *Module) {
+      if (isAvailableExternally(F.getLinkage()))
+        continue;
+
+      for (SILBasicBlock &BB : F) {
+        for (SILInstruction &I : BB) {
+          if (auto *FRI = dyn_cast<FunctionRefInst>(&I)) {
+            SILFunction *RefF = FRI->getReferencedFunction();
+            if (RefF->isTransparent() && RefF->isFragile())
+              ensureAlive(RefF);
```

Comment:
Why are these functions not reachable?

How does this interact with @inline(__always) or the future @_inlineable attribute? What makes @_transparent special here?